### PR TITLE
Refactoring .hpp -> .h, .cpp and remove class templates

### LINKED ISSUE #36: Remove class templates
The current implementation uses templates to define class float or double precision. There is no strong use-case for this and it makes the code cluttered and more complicated.

I will remove all of the class templates and just use double.

### PR DESCRIPTION
**Pull request by mrp089**
This is equivalent to #41 but leaves the file history intact. Closes #36.

**Do not squash and merge to leave history intact.**

### CODE REVIEW

**Comment by mrp089** [comment]
Steps to create new files while preserving the history:

1. Create `.cpp` files from `.hpp`
```
git mv FILE.hpp FILE.cpp
git commit
```

2. Reset `.hpp` files
```
SAVED=`git rev-parse HEAD`
git reset --hard HEAD^
```

3. Create `.h` files from `.hpp`
```
git mv FILE.hpp FILE.h
git commit
```

4. Merge commits to have both `.cpp` and `.h` files
```
git merge $SAVED
git commit -a
```

**Comment by mrp089** [comment]
I manually recreated the file structure in [@ktbolt's branch](https://github.com/ktbolt/svZeroDPlus/tree/Remove-class-templates_36) using `git mv`.

Since we can't do a squash and merge in this pull request, I already squashed @ktbolt's 21 commits into one using an interactive rebase (marking all but the first commit as `squash`):
```
git rebase -i HEAD~25
```

Finally, I applied @ktbolt's changes by creating a diff to the rebased branch with `git diff` and applied it as a patch with `git apply` to only change file contents but not the files themselves. I then commit that with @ktbolt as an author - the perfect crime!

**Comment by mrp089** [comment]
Note that there are still many files shown as newly created. However, this is just a `GitHub` problem (I assume it can't show two files as moved from the same original one).

If you do a `git blame` on one of the "newly created" files (e.g. [BloodVessel](https://github.com/StanfordCBCL/svZeroDPlus/blame/bbfdf71e099a282fd4f7f99689da8b1c000e65f5/src/model/BloodVessel.cpp)), you get the full history.

**Comment by mrp089** [comment]
@ktbolt, @menon-karthik I'll fix the Doxygen documentation. It doesn't find the files anymore and many comments need to be updated.

**Comment by mrp089** [comment]
Done! There are a few more Doxygen warnings, see #46, but nothing that should prevent this from being merged.

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,30 +57,51 @@ if(ENABLE_COVERAGE)
...
+# Set the location to store the binaries and libraries created by this project.
+# -----------------------------------------------------------------------------
+#
+#set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+#set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+#set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+# Need to add the -fPIC flag to build the interface shared library.
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
```
Comment (menon-karthik) [inline CMakeLists.txt:70]: @ktbolt Just wondering why this wasn't required previously?

File: src/model/Model.cpp [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "Model.h"
+
+#include "Node.h"
+
+namespace zd_model {
```
Comment (menon-karthik) [inline src/model/Model.cpp:35]: Do we really need to call this `zd_model` instead of just `model`?

File: src/model/FlowReferenceBC.cpp
Comment (menon-karthik) [inline src/model/FlowReferenceBC.cpp:1]: Git thinks many of these files are copied from `src/io/namespace`

**Review by menon-karthik — COMMENTED** [review]
Mostly looks good to me! I just have a few comments.

Thanks @ktbolt and @mrp089 for doing this!

File: src/model/FlowReferenceBC.cpp
Comment (menon-karthik) [reply src/model/FlowReferenceBC.cpp:1]: Just pointing this out, not saying it's something we should dig into.

Comment (mrp089) [reply src/model/FlowReferenceBC.cpp:1]: Yeah, that's weird. But I think it's just GitHub's diff display. In the history, [it shows up correctly](https://github.com/StanfordCBCL/svZeroDPlus/commits/70da0d6d17ab597f683a0c4d89d0a9f99dfb0656/src/model/flowreferencebc.hpp?browsing_rename_history=true&new_path=src/model/FlowReferenceBC.cpp&original_branch=6b716fdde4e54b3246dcec87a1ef34260c82e5e2):

File: src/model/Model.cpp (hunk 2 above)
Comment (ktbolt) [reply src/model/Model.cpp:35]: @menon-karthik Oops, meant to rename that, was testing for name collision problems. I plan on removing namespaces in a later Issue.

**Comment by ktbolt** [comment]
@mrp089 Thank you for getting all of this to work, my partner in crime!

**Pull request merged**